### PR TITLE
[doc] add a deployment guide and update the readme

### PR DESCRIPTION
**Pull request by sebsto**
As discussed with @0xTim 
This PR 
- adds a minimal example of deployment using SAM in the README 
- adds a `Deployment.md` Swift Docc file to cover deployment with the AWS console, CLI, SAM, and CDK. It mentions and contains a call to contributions to further examples for third-party tools such as the Serverless Framework, Terraform, or Pulumi.

### CODE REVIEW

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+      memorySize: 512,
+      timeout: cdk.Duration.seconds(30),
+      environment: {
+        LOG_LEVEL: 'debug',
+      },
+    });
+ }
+}    
+```
+The code assumes you already built the Swift Lambda function with the `swift package archive --allow-network-connections docker` command. The ZIP file is located in the `.build/plugins/AWSLambdaPackager/outputs/AWSLambdaPackager/MyLambda/MyLambda.zip` folder.
```
Comment (0xTim) [inline Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:474]: Again we should be clear about what app we're deploying here so there's no confusion when invoking the lambda etc

**Review by 0xTim — APPROVED** [review]
Overall it looks good. One typo and a few things to discuss

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+
+Select **Create function**
+
+![Console - create function](console-30-create-function)
+
+On the right side, select **Upload from** and select **.zip file**.
+
+![Console - select zip file](console-40-select-zip-file)
+
+Select the zip file created with the `swift package archive --allow-network-connections docker` command.  This file is located in your project folder at `.build/plugins/AWSLambdaPackager/outputs/AWSLambdaPackager/MyLambda/MyLambda.zip`. The name of the ZIP file depends on the target name you entered in the `Package.swift` file.
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:94]: This doc is about the deployment, so after the package is compiled and zipped.
But you're right, I should add a paragraph on top repeating the requirement : having a Swift lambda function compiled and ZIP.  This applies to all deployment techniques presented in this doc.

[the remark this answers is not in the document]
Comment (0xTim) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:94]: Yeah I'd add that before going into all the different examples

[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:94]: Done. I added a section in the prerequisites

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+
+Select **Save**
+
+![Console - select zip file](console-50-upload-zip)
+
+You're now ready to test your function.
+
+### Invoke the function 
+
+Select the **Test** tab in the console and prepare a payload to send to your Lambda function. In this example, you've deployed the [HelloWorld](Exmaples.HelloWorld/README.md) example function. The function expects a `String` as input parameter and returns a `String`.
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:104]: Done. I added a phrase at the start of the console section to describe the example we're going to deploy and a phrase here in the "Invoke" section to describe the test event

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+
+Select **Create new event**. Enter an **Event name**. Enter `"Swift on Lambda"` as **Event JSON**. Note that the payload must be a valid JSON document, hence we use surrounding double quotes (`"`).
+
+Select **Test** on the upper right side of the screen.
+
+![Console - prepare test event](console-60-prepare-test-event)
+
+The response of the invocation and additional meta data appears in the green section of the page.
+
+I can see the response from the Swift code: `Hello Swift on Lambda`.
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:114]: corrected

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+
+The function consumed 109.60ms of execution time, out of this 83.72ms where spent to initialize this new runtime. This initialization time is known as Lambda cold start time.
+
+> Lambda cold start time refers to the initial delay that occurs when a Lambda function is invoked for the first time or after being idle for a while. Cold starts happen because AWS needs to provision and initialize a new container, load your code, and start your runtime environment (in this case, the Swift runtime). This delay is particularly noticeable for the first invocation, but subsequent invocations (known as "warm starts") are typically much faster because the container and runtime are already initialized and ready to process requests. Cold starts are an important consideration when architecting serverless applications, especially for latency-sensitive workloads.
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:118]: > Usually, compiled languages, such as Swift, Go, and Rust, have shorter cold start times compared to interpreted languages, such as Python, Java, Ruby, and Node.js.


done

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 6]
```diff
@@ -0,0 +1,602 @@
...
+
+## Deploy your Lambda function with the AWS Command Line Interface (CLI)
+
+You can deploy your Lambda function using the AWS Command Line Interface (CLI). The CLI is a unified tool to manage your AWS services from the command line and automate your operations through scripts. The CLI is available for Windows, macOS, and Linux. Follow the [installation](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) and [configuration](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html) instructions in the AWS CLI User Guide.
+
+### Create the function 
+
+To create a function, you must first create the function execution role and define the permission. Then, you create the function with the `create-function` command.
+
+The command assumes you've already created the ZIP file with the `swift package archive --allow-network-connections docker` command. The name and the path of the ZIP file depends on the executable target name you entered in the `Package.swift` file.
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:154]: > In this example, we're building the HelloWorld example from the [Examples](https://github.com/swift-server/swift-aws-lambda-runtime/tree/main/Examples) directory.

done

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 7]
```diff
@@ -0,0 +1,602 @@
...
+--zip-file fileb://.build/plugins/AWSLambdaPackager/outputs/AWSLambdaPackager/MyLambda/MyLambda.zip \
+--runtime provided.al2 \
+--handler provided  \
+--architectures arm64 \
+--role arn:aws:iam::${AWS_ACCOUNT_ID}:role/lambda_basic_execution
+```
+
+The `--architectures` flag is only required when you build the binary on an Apple Silicon machine (Apple M1 or more recent). It defaults to `x64`.
+
+To update the function, use the `update-function-code` command.
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:218]: > To update the function, use the `update-function-code` command after you've recompiled and archived your code again with the `swift package archive` command.

done

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 8]
```diff
@@ -0,0 +1,602 @@
...
+  # display API Gateway endpoint
+  APIGatewayEndpoint:
+    Description: "API Gateway endpoint URI"
+    Value: !Sub "https://${ServerlessHttpApi}.execute-api.${AWS::Region}.amazonaws.com"
+EOF
+```
+
+In this example, the Lambda function must accept an APIGateway v2 JSON payload as input parameter and return a valid APIGAteway v2 JSON response.  See the example code in the [APIGateway example README file](https://github.com/swift-server/swift-aws-lambda-runtime/blob/main/Examples/APIGateway/README.md).
+
+To deploy the function with SAM, use the `sam deploy` command. The very first time you deploy a function, you must use the `--guided` flag to configure the deployment. The command will ask you a series of questions to configure the deployment.
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:307]: done

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md [hunk 9]
```diff
@@ -0,0 +1,602 @@
...
+        Looking for resources needed for deployment:
+
+(redacted for brevity)
+
+CloudFormation outputs from deployed stack
+--------------------------------------------------------------------------------
+Outputs                                                                                                                                         
+--------------------------------------------------------------------------------
+Key                 APIGatewayEndpoint                                                                                                          
+Description         API Gateway endpoint UR"
```
[the remark this answers is not in the document]
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:345]: done

File: Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md (hunk 1 above)
Comment (sebsto) [reply Sources/AWSLambdaRuntimeCore/Documentation.docc/Deployment.md:474]: done, I added a phrase at the start of the section and clarified this one.

**Comment by sebsto** [comment]
Thank you @0xTim for having taken the time to review this, I included all modifications you suggested.
You can see the list of changes in this commit https://github.com/swift-server/swift-aws-lambda-runtime/pull/432/commits/17f6b46b914bc3036e18dd181b5c03c53cae85b4

**Review by 0xTim — APPROVED** [review]
LGTM!

**Pull request merged**